ipn/ipnserver/server.go

```diff
@@ -1114,3 +1126,17 @@ func (ln *listenerWithReadyConn) Accept() (net.Conn, error) {
 	}
 	return ln.Listener.Accept()
 }
+
+// findSynologyTaildropDir looks for the first volume containing a
+// "Taildrop" directory.  We'd run "synoshare --get Taildrop" command
+// but on DSM7 at least, we lack permissions to run that.
+func findSynologyTaildropDir() (dir string, err error) {
+	const name = "Taildrop"
+	for i := 1; i <= 16; i++ {
+		dir = fmt.Sprintf("/volume%v/%s", i, name)
```

Comment:
The path to the tailscale application has turned out to vary depending on whether the Synology system was first installed with DSM6/7 or if it started out life with a much earlier version and was repeatedly upgraded. Synology seems to keep putting app files in the location where that really old version first put them.

Do we have any reason to believe that volume handling would be similar, maybe different on some devices?

I guess the best way to proceed is ship it and see if there are systems where it doesn't work.

Reply:
> Do we have any reason to believe that volume handling would be similar, maybe different on some devices?

I've never seen anything but /volumeN/$NAME on any of my Synology devices on either DSM 6 or 7.

But nothing surprises me anymore. Wait and see I guess?

It tried to do it the "right" way but I got access denied querying that info, so not sure what else to do.